hack/ginkgo-e2e.sh

```diff
@@ -60,13 +60,6 @@ if [[ -z "${AUTH_CONFIG:-}" ]];  then
       auth_config=(
       "--kubeconfig=${KUBECONFIG:-$DEFAULT_KUBECONFIG}"
       )
-      if [[ "${KUBERNETES_PROVIDER}" == "gke" ]]; then
-        # gcloud doesn't override the current-context, so we explicitly set it
-        detect-project &> /dev/null
```

Comment:
Not needed. Tests that depend on zone/project should be calling `detect-project` themselves.